src/parser.rs

```diff
@@ -0,0 +1,219 @@
+use crate::ast::{Identifier, PluginBuilder, PluginOption, Tag, Task, TaskBuilder, Value};
+use pest::iterators::Pairs;
+use pest_derive::Parser;
+
+#[derive(Parser)]
+#[grammar = "dsl.pest"]
+struct DSLParser;
+
+fn create_task(pairs: Pairs<Rule>, task_builder: &mut TaskBuilder) {
+    let mut tags = Vec::new();
+    for pair in pairs {
+        match pair.as_rule() {
+            Rule::task_name => {

```

Comment:
Shouldn't we assert that we first get a name, followed by a curly, and only then a plugin?

I wrote this test and was surprised that it passes:

```rust
    #[test]
    fn test_foo() {
        let input = r#"
    task  
        foo {}
    "#;

        let pairs = DSLParser::parse(Rule::tasks, input).expect("Parsable dsl");
        build_ast(pairs, 0);
    }

```